06_neighborhoods/environment/air_toxicity_metric.qmd

```diff
@@ -0,0 +1,855 @@
+  mutate(county = str_sub(county, start = -3L, end = -1L),
+         state = str_pad(state, 2, side = "left", pad = "0"),
+         tract = tract * 100,
+         tract = str_pad(tract, 6, side = "left", pad = "0"),
+         tract = paste0(state, county, tract),
+         year1 = "2019",
+         year2 = "2018",
+         year3 = "2014") %>% 
+  select(place = placefp, tract, afact,  year1:year3) %>% 
+  pivot_longer(!c(place, tract, afact), values_to = "year", names_to = "value") %>% 
+  select(-value)
+
+```
+
+Pad the place variable with a leading zero.
+```{r}
+crosswalk_place <- crosswalk_place %>% 
+  mutate(place = str_pad(place, 5, side = "left", pad = "0")) %>% 
+  select(year, place, tract, afact)
+```
+
+Merge the environmental index ACS data onto the city crosswalk file.
+```{r}
+tract_place_hazard <- crosswalk_place %>% 
+  left_join(haz_index_all, by = c("year", "tract")) %>%
+  mutate(place = as.character(place),
+         state = str_sub(tract, 1, 2)) %>% 
+  mutate(place_id = str_c(state, place))
+
+anti_join(crosswalk_place, haz_index_all, by = c("year", "tract")) %>% 
+  mutate(state = str_sub(tract, start = 1, end = 2)) %>% 
+  group_by(state, year) %>%
+  count() %>% 
+  arrange(desc(n))
+```
+
+Merge on ACS information.
+```{r}
+haz_index_acs <- tract_place_hazard %>% 
+  left_join(race_pov, by=c("tract" = "geoid", "year")) %>% 
+  mutate(na_pop= if_else(is.na(index_air_hazard), total_pop, 0))
+```
+
+## Create place level index
+
+Import places/pop file and prep to limit to population cutoff places.
+```{r}
+state_place_pop <- read_csv(here::here("geographic-crosswalks", "data", "place-populations.csv")) %>% 
+  mutate(place_id = paste0(state, place))
+```
+
+Create "All" observations calculate place population for tracts and index_air_hazard weighting by amount of tract in place and tract total population
+
+```{r}
+all_enviro_place <- haz_index_acs %>%
+  mutate(tract_pop = (total_pop*afact)) %>% #to account for the fact that only part of the tract population is in the place
+  group_by(year, place_id, state, place) %>%
+  summarise(index_air_hazard = weighted.mean(index_air_hazard, tract_pop, na.rm = TRUE),
+        na_pop = sum(na_pop * afact),
+        place_pop = sum (tract_pop)) %>% 
+        ungroup()
+```
+
+Add a variable with percent of population missing for each place
+
+```{r}
+all_enviro_place <- all_enviro_place %>%
+  mutate(na_perc = na_pop / place_pop,
+      subgroup = "All",
+      subgroup_type = "all") %>%
+select(-c(na_pop, place_pop))
+```
+### Poverty level
+
+Create place level weighted index by poverty level.
+```{r}
+hazard_place_pov <- haz_index_acs %>%
+  mutate(na_pop = if_else(is.na(index_air_hazard) | is.na(poverty_type), total_pop, 0)) 
+```
+Create the index in each place by the poverty type subgroup.
+```{r}
+hazard_place_sub_pov <- hazard_place_pov %>%
+  group_by(year, place_id, state, place, poverty_type) %>%
```

Comment:
Here you are multiplying `total_pop*afact` for the weight which I assume is the correct way to do this. Please make sure this is consistently applied in each summarise step.

Reply:
To the previous comment it seems like you can either do `total_pop*afact` or `tract_pop` which is the total_pop already adjusted.

Reply:
For consistency I'll stick to the tract_pop approach.